zerocopy-derive/tests/ui-nightly/struct.rs

```diff
@@ -23,15 +22,31 @@ fn main() {}
 // KnownLayout errors
 //
 
+struct NotKnownLayout;
+
+struct NotKnownLayoutDst([u8]);
+
+// | `repr(C)`? | generic? | `KnownLayout`? | `Sized`? | Type Name |
+// |          N |        N |              N |        N |      KL00 |
 #[derive(KnownLayout)]
-struct KnownLayout1([u8]);
+struct KL00(u8, NotKnownLayoutDst);
 
-assert_impl_all!(KnownLayout1: KnownLayout);
+// | `repr(C)`? | generic? | `KnownLayout`? | `Sized`? | Type Name |
+// |          N |        N |              Y |        N |      KL02 |
+#[derive(KnownLayout)]
+struct KL02(u8, [u8]);
 
+// | `repr(C)`? | generic? | `KnownLayout`? | `Sized`? | Type Name |
+// |          Y |        N |              N |        N |      KL08 |
 #[derive(KnownLayout)]
-struct KnownLayout2<T: ?Sized>(T);
+#[repr(C)]
+struct KL08(u8, NotKnownLayoutDst);
 
-assert_impl_all!(KnownLayout2<[u8]>: KnownLayout);
+// | `repr(C)`? | generic? | `KnownLayout`? | `Sized`? | Type Name |
+// |          Y |        N |              N |        Y |      KL09 |
+#[derive(KnownLayout)]
+#[repr(C)]
+struct KL09(NotKnownLayout, NotKnownLayout);
 
```

Comment:
Since this succeeds, should it be in a non-UI test file?

Reply:
It doesn't succeed. It produces this error:
```
error[E0277]: the trait bound `NotKnownLayout: KnownLayout` is not satisfied
  --> tests/ui-nightly/struct.rs:47:10
   |
47 | #[derive(KnownLayout)]
   |          ^^^^^^^^^^^ the trait `KnownLayout` is not implemented for `NotKnownLayout`
   |
   = help: the following other types implement trait `KnownLayout`:
             bool
             char
             isize
             i8
             i16
             i32
             i64
             i128
           and $N others
   = help: see issue #48214
   = help: add `#![feature(trivial_bounds)]` to the crate attributes to enable
   = note: this error originates in the derive macro `KnownLayout` (in Nightly builds, run with -Z macro-backtrace for more info)
```

Reply:
Ah gotcha; I was cmd+f-ing for `KL09`, which doesn't appear in that error output. My bad.